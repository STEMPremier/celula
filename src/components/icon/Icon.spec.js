import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { ProductIcon, SecondaryIcon, SystemIcon } from '.';

describe('<ProductIcon />', () => {
  describe('render', () => {
    test('renders the component', () => {
      const wrapper = shallow(<ProductIcon name="music" />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});

describe('<SecondaryIcon />', () => {
  describe('render', () => {
    test('renders the component with an icon', () => {
      const wrapper = shallow(<SecondaryIcon name="image" />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('renders the component with text', () => {
      const wrapper = shallow(<SecondaryIcon text="+1" />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});

describe('<SystemIcon />', () => {
  describe('render', () => {
    test('renders the component', () => {
      const wrapper = shallow(<SystemIcon name="share" />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
