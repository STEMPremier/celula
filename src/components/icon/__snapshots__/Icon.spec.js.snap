// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductIcon /> render renders the component 1`] = `
<span
  className="ce-icon ce-icon__product ce-icon--small"
>
  <svg
    className="ce-icon--white"
  >
    <use
      xlinkHref="#icons_product-music"
    />
  </svg>
</span>
`;

exports[`<SecondaryIcon /> render renders the component with an icon 1`] = `
<span
  className="ce-icon ce-icon__secondary ce-icon--small ce-icon--primary"
>
  <svg
    className="ce-icon--white"
  >
    <use
      xlinkHref="#icons_secondary-image"
    />
  </svg>
</span>
`;

exports[`<SecondaryIcon /> render renders the component with text 1`] = `
<span
  className="ce-icon ce-icon__secondary ce-icon--small ce-icon--primary"
>
  <span
    className="ce-icon--text"
  >
    +1
  </span>
</span>
`;

exports[`<SystemIcon /> render renders the component 1`] = `
<span
  className="ce-icon ce-icon__system"
>
  <svg
    className="ce-icon--black"
  >
    <use
      xlinkHref="#icons_system-share"
    />
  </svg>
</span>
`;
